server/configuration.go

```diff
@@ -128,12 +128,29 @@ func (p *Plugin) OnConfigurationChange() error {
 	if err := p.API.LoadPluginConfiguration(configuration); err != nil {
 		return errors.Wrap(err, "failed to load plugin configuration")
 	}
+	if _, err := isValidAuthConfig(configuration); err != nil {
 
-	if configuration.EnableLegacyAuth && configuration.EnableOAuth {
-		return errors.New("Only enable One of the OAuth or Password based authentication")
+		if apiErr := p.API.DisablePlugin(PLUGIN_ID); apiErr != nil {
```

Comment:
@levb please suggest if this is a reasonable thing to disable plugin when invalid config is detected on config change

Reply:
1/5 I suppose there is another choice (but a separate PR for sure) - which is, when config is invalid, fail all operations like starting meetings, and send the user an ephemeral message saying that the plugin is misconfigured, contact sysadmin. Other plugins (used to?) do it. @aaronrothschild

Reply:
I think the webex plugin does something like that.